nengo_loihi/compat.py

```diff
@@ -0,0 +1,38 @@
+from distutils.version import LooseVersion
+
+import nengo
+import numpy as np
+
+if LooseVersion(nengo.__version__) > LooseVersion('2.8.0'):
+    import nengo.transforms as nengo_transforms
+
+    def conn_solver(solver, activities, targets, rng):
+        return solver(activities, targets, rng=rng)
+
+    def transform_array(transform):
+        return transform.init
+
+    def sample_transform(conn, rng=np.random):
+        return conn.transform.sample(rng=rng)
+
+else:
+    nengo_transforms = None
+    from nengo.dists import get_samples as _get_samples
+
+    nengo.solvers.Solver.compositional = True
+    nengo.solvers.LstsqDrop.compositional = False
+    nengo.solvers.LstsqL1.compositional = False
+    nengo.solvers.Nnls.compositional = False
```

Comment:
One danger to this is it would affect other packages/places someone is using nengo. It's quite unlikely that that would actually be a problem, but I do always worry about monkey patching like this. It does make things much cleaner, though, so I like that aspect.

Reply:
Yeah, in this case since this is only an internal builder attribute (so we know how and where it is being used), the risk seemed low enough to me.  The only case this would cause a problem is if a) someone is writing their own code that uses the compositional attribute (writing their own build functions or something?), and b) they don't want those attributes to have the values defined in the solver (for some reason).  And this person has to be using an old version of nengo, where the `compositional` attribute didn't even exist.